src/edgeProxy/public/javascripts/uio.js

```diff
@@ -1,19 +1,166 @@
+/* global getCookieValue */
+
 "use strict";
 
-fluid.uiOptions(".flc-prefsEditor-separatedPanel", {
-    auxiliarySchema: {
-        terms: {
-            "templatePrefix": "lib/infusion/src/framework/preferences/html",
-            "messagePrefix": "lib/infusion/src/framework/preferences/messages"
-        },
-        "fluid.prefs.tableOfContents": {
-            enactor: {
-                "tocTemplate": "lib/infusion/src/components/tableOfContents/html/TableOfContents.html",
-                "tocMessage": "lib/infusion/src/framework/preferences/messages/tableOfContents-enactor.json",
-                ignoreForToC: {
-                    "overviewPanel": ".flc-overviewPanel"
-                }
+fluid.registerNamespace("fluid.prefs.edgeProxy");
+
+fluid.defaults("fluid.dataSource.noEncoding", {
+    gradeNames: ["fluid.dataSource", "fluid.dataSource.writable"],
+    components: {
+        encoding: {
+            type: "fluid.dataSource.encoding.none"
+        }
+    }
+});
+
+// Edge Proxy Store
+fluid.defaults("fluid.prefs.edgeProxyStore", {
+    gradeNames: ["fluid.dataSource.noEncoding", "fluid.modelComponent"],
+    members: {
+        isFreshLogon: "@expand:fluid.prefs.edgeProxyStore.getFreshLogon()"
+    },
+    model: {
+        isLoggedIn: false
+    },
+    components: {
+        unauthedStore: {
+            type: "fluid.prefs.cookieStore",
+            options: {
+                writable: true
             }
         }
+    },
+    dynamicComponents: {
+        authedStore: {
+            source: "{that}.model.isLoggedIn",
+            type: "fluid.prefs.pdsStore"
+        }
+    },
+    listeners: {
+        "onCreate.updateLoggedInState": {
+            listener: "fluid.prefs.edgeProxyStore.updateLoggedInState",
+            args: ["{that}"]
+        },
+        "onRead.impl": {
+            listener: "fluid.prefs.edgeProxyStore.get",
+            args: ["{that}"]
+        },
+        "onWrite.impl": {
+            listener: "fluid.prefs.edgeProxyStore.set",
+            args: ["{that}", "{arguments}.0"]   // settings
+        }
+    },
+    modelListeners: {
+        isLoggedIn: {
+            listener: "fluid.prefs.edgeProxyStore.updateUnauthedSettings",
+            args: ["{that}", "{prefsEditorLoader}", "{change}.value"],
+            excludeSource: "init"
+        }
     }
 });
+
+// Update the fresh logon state at the page load. Check if PDS_freshLogon is a query parameter of the page url.
+// If yes, return true. Otherwise, return false.
+fluid.prefs.edgeProxyStore.getFreshLogon = function () {
+    const url = new URL(window.location);
+    const isFreshLogon = url.searchParams.get("PDS_freshLogon") === "true";
```

Comment:
I suppose this doesn't guarantee that the token is valid. Does that matter at this point?

Reply:
> I suppose this doesn't guarantee that the token is valid. Does that matter at this point?

When "PDS_freshLogon" is not in query strings, `url.searchParams.get("PDS_freshLogon")` returns `null` and `isFreshLogon` will still be correctly set to `false`.

Reply:
I guess I mean if the token is present but not valid, e.g. fraudulent, expired, revoked, etc., it will still assume the user is logged in when they aren't. I guess it might depend on what error handling is available for fetching/saving data to the PDS.

Reply:
When a login token is invalid, Personal Data Server API will return an error response. We should talk about the error handling with an UIO store. My questions are:
1. Shall the error be escalated to be handled out of UIO?
or,
2. Shall UIO provide an option to accept error handling function(s)?